config/plugins/interactive_environments/rstudio/templates/rstudio.mako

```diff
@@ -0,0 +1,84 @@
+<%namespace file="ie.mako" name="ie"/>
+<%
+import os
+import shutil
+import tempfile
+import time
+import subprocess
+
+# Sets ID and sets up a lot of other variables
+ie_request.load_deploy_config()
+ie_request.attr.docker_port = 80
+# Create tempdir in galaxy
+temp_dir = os.path.abspath( tempfile.mkdtemp() )
+# We have to do some special things with the password here. Currently there's
+# an unpatched issue in Galaxy-IE which means passwords are automatically hashed
+# according to what IPython expects (sha1+salt). Unfortunately this is also done
+# for RStudio which takes the password as-is. However, we give the IE plugin NO
+# way to access this hashed password, so we have to work around this. Fortunately
+# we placed the additional conf entries at the very end of of the
+# `write_conf_file` function, enabling us to overwrite the correct attributes
+# simply by passing `notebook_password` with a "plaintext", unhashed password.
+PASSWORD = ie_request.generate_password(length=36)
+USERNAME = "galaxy"
+# Write out conf file...needs work
+ie_request.write_conf_file(temp_dir, {'notebook_username': 'galaxy', 'notebook_password': PASSWORD,
+                                      'cors_origin': ie_request.attr.proxy_url})
+# This is overwritten at the end of the above function call, so we need to re-overwrite it.
+ie_request.attr.notebook_pw = PASSWORD
+
+## General IE specific
+# Access URLs for the notebook from within galaxy.
+# TODO: Make this work without pointing directly to IE. Currently does not work
+# through proxy.
+notebook_pubkey_url = ie_request.url_template('${PROXY_URL}/rstudio/${PORT}/auth-public-key')
+notebook_access_url = ie_request.url_template('${PROXY_URL}/rstudio/${PORT}/')
+notebook_login_url =  ie_request.url_template('${PROXY_URL}/rstudio/${PORT}/auth-do-sign-in')
+
+# Did the user give us an RData file?
+if hda.datatype.__class__.__name__ == "RData":
```

Comment:
Huh. That's actually a good question, I hadn't given that any thought.

`.RData` files should just be .gz of data structures loaded in the R session. I don't know think anything is included in the IE that wouldn't be able to be used outside of the IE, so I'd lean towards keeping it as-is, just raw RData files. I'll double check this tomorrow, that the .RData files are completely functional inside and out. If they are, I'd say no reason to impose any extraneous limits.